src/io/flutter/utils/EventStream.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.utils;
+
+
+import com.intellij.openapi.application.ApplicationManager;
+
+import javax.swing.*;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Simple class for listening for a value that can be repeatedly updated.
+ * <p>
+ * The  benefit of this class instead of listening for events is you don't have
+ * to worry about missing values posted to the stream before your code started
+ * listening.
+ * The benefit of using this class over awaiting a Future is that the value can
+ * update multiple times.
+ * <p>
+ * New values to the EventStream can be posted on any thread to listen can be
+ * made from any thread.
+ * <p>
+ * The class is inspired by listen method on the Stream class in Dart.
+ */
+public class EventStream<T> {
+
+  protected final HashSet<StreamSubscription<T>> subscriptions = new LinkedHashSet<>();
+
+  private volatile T currentValue;
+
+  public EventStream(T initialValue) {
```

Comment:
Perhaps also have a 

```
public EventStream() {
  currentValue = null;
}
```

constructor (and use this in the call sites that currently pass in null)?

Reply:
done.